pytd/client.py

```diff
@@ -119,7 +120,7 @@ def close(self):
         """
         self.engine.close()
         self.api_client.close()
-        if self.writer is not None:
+        if self.writer is not None and self.managed_writer:
```

Comment:
Since `self.writer is not None` is always `True` where `self.managed_writer` is `False`, I personally prefer following order:

```py
if self.managed_writer and self.writer is not None:
```

For unmanaged writer, the `if` statement quickly realizes the condition becomes `False` by only checking the first sub-condition, `self.managed_writer`; the second comparison `self.writer is not None` can be skipped in the suggested ordering.

Reply:
Agree. fixed 80f36ae